### PR TITLE
Tickets/dm 33411 Fix VCD and add status to verification element in report

### PR DESCRIPTION
**Pull request by womullan**
Fix VCD and add status to verification element in report

### CODE REVIEW

File: docsteady.py [hunk 1]
```diff
@@ -2,5 +2,8 @@
 from docsteady import cli
 import sys
 
+import warnings
+warnings.filterwarnings("ignore", category=UserWarning, module='bs4')
```
Comment (timj) [inline docsteady.py:6]: Can you add a comment here explaining why you need to do this? Also, in general we try to be consistent with quoting -- single vs double -- within a file. In theory turning on black would fix all of this.

File: docsteady/__init__.py [hunk 2]
```diff
@@ -368,7 +368,14 @@ def generate_vcd(format, jiradb, vcduser, vcdpwd, username, password, sql, spec,
             if 'req_doc_id' in ve_model[ve].keys():
                 req_dict[ve_model[ve]['req_id']]['reqDoc'] = ve_model[ve]['req_doc_id']
             ve_dict[ve_long_name[0]] = tmp_ve
-        vcd_dict = [ve_dict, req_dict, [], Config.CACHED_TESTCASES]
+        # Not sure why the ve_dict is keyed on Requirement with a version -
+        # everying wants verificaiton element so remaking it (wom)
```
Comment (timj) [inline docsteady/__init__.py:372]: typo: "everying"

You don't need to sign a comment -- git blame will tell us who made it.

File: docsteady/__init__.py [hunk 3]
```diff
@@ -368,7 +368,14 @@ def generate_vcd(format, jiradb, vcduser, vcdpwd, username, password, sql, spec,
...
+        vee_dict = {}
```
Comment (timj) [inline docsteady/__init__.py:373]: The variable names are a bit opaque -- I assume it's obvious how ve_dict and vee_dict are related?

File: docsteady/__init__.py [hunk 4]
```diff
@@ -368,7 +368,14 @@ def generate_vcd(format, jiradb, vcduser, vcdpwd, username, password, sql, spec,
...
+        for vreq, elem in ve_dict.items():
+            lvv = elem['jkey']
+            vee_dict[lvv] = elem
+        # now keyd on verification element it should work in jinga
```
Comment (timj) [inline docsteady/__init__.py:377]: Typo: keyd

File: docsteady/vcd.py [hunk 5]
```diff
@@ -425,7 +428,8 @@ def get_ves(comp):
                     tsum = vby[1].split(':')
                     vbytmp.append(tsum[0])
                     verifying_ves.append(str(vby[0]))
-                tmpve['verifiedby'] = vbytmp
+                tmpve['verified_by'] = vbytmp  # was without _
```
Comment (timj) [inline docsteady/vcd.py:431]: If the comment is needed can it say what caused the transition from verifiedby to verified_by?

File: docsteady/vcd.py [hunk 6]
```diff
@@ -553,14 +557,18 @@ def do_req_coverage(ves, ve_coverage):
     """
     Calculate the coverage level of a requirement
     based on the downstram verification elements.
+    :param myvreq: version requirement name
```
Comment (timj) [inline docsteady/vcd.py:560]: Eventually these should be changed to numpydoc everywhere so we can integrate into our sphinx system.

File: docsteady/vcd.py [hunk 7]
```diff
@@ -553,14 +557,18 @@ def do_req_coverage(ves, ve_coverage):
...
     :param ves:
     :param ve_coverage:
     :return:
     """
     nves = len(ves)
     vecount = Counter()
     for ve in ves:
-        vecount.update([ve_coverage[ve]['coverage']])
+        element = ve_coverage[ve]
+        # there is only one VE per requirement ?
```
Comment (timj) [inline docsteady/vcd.py:569]: Is that a question for the person reading the code or is it saying that the value of `cover` indicated where there is one VE per requirement?

File: docsteady/vcd.py [hunk 8]
```diff
@@ -579,13 +587,19 @@ def do_req_coverage(ves, ve_coverage):
     return rcoverage
 
 
+def find_vekey(reqname, ve_keys):
+    """ Look thorugh the keys until i find the one my equiement starts with"""
```
Comment (timj) [inline docsteady/vcd.py:591]: Typo: `thorugh` and `i`, although "I" is not usual for docstrings.

File: docsteady/vcd.py [hunk 9]
```diff
@@ -594,27 +608,29 @@ def summary(dictionary):
     mtrs['nv'] = len(verification_elements)
     mtrs['nt'] = len(tcases)
 
-    for req in dictionary[1].values():
+    for reqname, req in reqs.items():
         Config.REQ_STATUS_PER_DOC_COUNT.update([req["reqDoc"]])
         Config.REQ_STATUS_PER_DOC_COUNT.update([req["reqDoc"]+"."+req["priority"]])
-        # print('VE - KEYS ', dictionary[0].keys())
+        # all this had verifiedby no _
```
Comment (timj) [inline docsteady/vcd.py:614]: Is the comment necessary for the current version of the code?

**Review by timj — APPROVED** [review]
Some minor comments.

File: docsteady/__init__.py (hunk 3 above)
Comment (womullan) [reply docsteady/__init__.py:373]: comment added

File: docsteady/vcd.py [hunk 10]
```diff
@@ -65,7 +65,10 @@ def extract_fields(self, data):
         data["ve_status"] = data_fields["status"]["name"]
         if data_fields["priority"]:
             data["ve_priority"] = data_fields["priority"]["name"]
-        data["req_id"] = data_fields["customfield_15502"]
+        try:
+            data["req_id"] = data_fields["customfield_15502"]
```
[the remark this answers is not in the document]
Comment (womullan) [reply docsteady/vcd.py:69]: no only if it exists its used in the template - if it does not exist you get an exception here. Apparently this always existed before and now does not always exist ..

File: docsteady/__init__.py (hunk 2 above)
Comment (timj) [inline docsteady/__init__.py:372]: Forgot to mention "verificaiton" as well.

File: docsteady/vcd.py (hunk 5 above)
Comment (womullan) [reply docsteady/vcd.py:431]: I wish I knew - I have no idea how this change happened - perhaps someine renamed it in Jira ....  it worked last year

File: docsteady/vcd.py (hunk 6 above)
Comment (womullan) [reply docsteady/vcd.py:560]: i guess we could make a ticket

File: docsteady/vcd.py (hunk 7 above)
Comment (womullan) [reply docsteady/vcd.py:569]: I guess it was my comment looking at the code - it seems to be implied but its not commented and theoretically may not be true .. I will update

**Pull request merged**